Latex and Sphinx are not yet functioning together.

This example is working on sphinx:
```
.. manim:: DotScene1
    :quality: low
    :save_last_frame:

        class DotScene1(Scene):
            def construct(self):
                dot = Dot().set_color(GREEN)
                self.add(dot)
                self.wait(1)
```
However, when I have something tex related, it throws an error:
```
.. manim:: TextExample
    :quality: medium
    :save_last_frame:
    
    class TextExample(Scene):
        def construct(self):
            t = TextMobject("Hello World")
            self.add(t)
```
> Exception occurred:
  File "/home/k/projects/manim-community/manim/utils/tex_file_writing.py", line 32, in generate_tex_file
    with open(result, "w", encoding="utf-8") as outfile:
FileNotFoundError: [Errno 2] No such file or directory: 'media/Tex/7d1ec941f0e30957.tex'
The full traceback has been saved in /tmp/sphinx-err-4zdxhjgt.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:26: html] Error 2

A similar error message comes e.g. for this example:
```
.. manim:: Plot1
    :quality: medium
    :save_last_frame:

    class Plot1(GraphScene):
       def construct(self):
           self.setup_axes()
           my_func = lambda x: np.sin(x)
           func_graph=self.get_graph(my_func)
           self.add(func_graph)
```

--- a/docs/source/manim_directive.py
+++ b/docs/source/manim_directive.py
@@ -153,7 +153,11 @@ class ManimDirective(Directive):
         source_block = "\n".join(source_block)
 
         media_dir = os.path.join(setup.confdir, "media")
+        if not os.path.exists(media_dir):
+            os.mkdir(media_dir)
         images_dir = os.path.join(media_dir, "images")
+        if not os.path.exists(images_dir):
+            os.mkdir(images_dir)
         tex_dir = os.path.join(media_dir, "tex")
         if not os.path.exists(tex_dir):
             os.mkdir(tex_dir)

--- a/docs/source/manim_directive.py
+++ b/docs/source/manim_directive.py
@@ -152,7 +152,7 @@ class ManimDirective(Directive):
         ]
         source_block = "\n".join(source_block)
 
-        media_dir = os.path.join("source", "media")
+        media_dir = os.path.join(setup.confdir, "media")
         images_dir = os.path.join(media_dir, "images")
         video_dir = os.path.join(media_dir, "videos")
         output_file = f"{clsname}-{classnamedict[clsname]}"
@@ -200,7 +200,7 @@ class ManimDirective(Directive):
 
         rendered_template = jinja2.Template(TEMPLATE).render(
             display_source=display_source,
-            filesrc=filesrc[6:],
+            filesrc_rel=os.path.relpath(filesrc, setup.confdir),
             output_file=output_file,
             save_last_frame=save_last_frame,
             save_as_gif=save_as_gif,
@@ -239,10 +239,10 @@ TEMPLATE = r"""
 
     <video class="manim-video" controls loop autoplay src="./{{ output_file }}.mp4"></video>
 {% elif save_as_gif %}
-.. image:: {{ filesrc }}
+.. image:: /{{ filesrc_rel }}
     :align: center
 {% elif save_last_frame %}
-.. image:: {{ filesrc }}
+.. image:: /{{ filesrc_rel }}
     :align: center
 {% endif %}
 

--- a/docs/source/manim_directive.py
+++ b/docs/source/manim_directive.py
@@ -154,6 +154,12 @@ class ManimDirective(Directive):
 
         media_dir = os.path.join(setup.confdir, "media")
         images_dir = os.path.join(media_dir, "images")
+        tex_dir = os.path.join(media_dir, "tex")
+        if not os.path.exists(tex_dir):
+            os.mkdir(tex_dir)
+        text_dir = os.path.join(media_dir, "text")
+        if not os.path.exists(text_dir):
+            os.mkdir(text_dir)
         video_dir = os.path.join(media_dir, "videos")
         output_file = f"{clsname}-{classnamedict[clsname]}"
 
@@ -163,6 +169,8 @@ class ManimDirective(Directive):
             f'config["pixel_width"] = {pixel_width}',
             f'file_writer_config["media_dir"] = "{media_dir}"',
             f'file_writer_config["images_dir"] = "{images_dir}"',
+            f'file_writer_config["tex_dir"] = "{tex_dir}"',
+            f'file_writer_config["text_dir"] = "{text_dir}"',
             f'file_writer_config["video_dir"] = "{video_dir}"',
             f'file_writer_config["save_last_frame"] = {save_last_frame}',
             f'file_writer_config["save_as_gif"] = {save_as_gif}',

